src/go/configinfo/service_info.go

```diff
@@ -47,29 +47,30 @@ type ServiceInfo struct {
 	Operations []string
 	// Stores all methods info for this service, using selector as key.
 	Methods map[string]*methodInfo
-	// Stores information about backend clusters for re-routing.
-	BackendRoutingClusters []*BackendRoutingCluster
 	// Stores url segment names, mapping snake name to Json name.
 	SegmentNames []*pmpb.SegmentName
 
 	AllowCors         bool
-	BackendIsGrpc     bool
 	ServiceControlURI string
-	CatchAllBackend   *BackendRoutingCluster
 	GcpAttributes     *scpb.GcpAttributes
 	// Keep a pointer to original service config. Should always process rules
 	// inside ServiceInfo.
 	serviceConfig *confpb.Service
 	AccessToken   *commonpb.AccessToken
 	Options       options.ConfigGeneratorOptions
+
+	// Stores information about all backend clusters.
+	AnyBackendIsGrpc       bool
+	CatchAllBackend        *BackendRoutingCluster
+	BackendRoutingClusters []*BackendRoutingCluster
 }
 
 type BackendRoutingCluster struct {
 	ClusterName  string
 	Hostname     string
 	Port         uint32
 	UseTLS       bool
-	Protocol     util.BackendProtocol
+	Protocol     util.Protocol
 	HttpProtocol util.HttpProtocol
```

Comment:
Do we really need two protocol here?  We can definitely unify them.

Reply:
gRPC is HTTP2,  and you don't need gRPC anymore, when you have this "AnyBackendIsGrpc",

Reply:
Discussed offline with @qiwzhang, seems fine. Done